pkg/handlers/generic/mutation/controlplanevirtualip/inject_test.go

```diff
@@ -52,28 +60,24 @@ var _ = Describe("Generate ControlPlane virtual IP patches", func() {
 						VariableName,
 					),
 				},
-				RequestItem: request.NewKubeadmControlPlaneTemplateRequestItem(""),
+				RequestItem: requestItem,
 				ExpectedPatchMatchers: []capitest.JSONPatchMatcher{
 					{
-						Operation: "add",
-						Path:      "/spec/template/spec/kubeadmConfigSpec/files",
-						ValueMatcher: gomega.ContainElements(
-							gomega.SatisfyAll(
-								gomega.HaveKeyWithValue(
-									"content",
-									gomega.ContainSubstring("value: \"10.20.100.10\""),
-								),
-								gomega.HaveKeyWithValue(
-									"content",
-									gomega.ContainSubstring("value: \"6443\""),
-								),
-								gomega.HaveKey("owner"),
-								gomega.HaveKeyWithValue(
-									"path",
-									gomega.ContainSubstring("kube-vip"),
-								),
-								gomega.HaveKey("permissions"),
-							),
+						Operation:    "add",
+						Path:         "/spec/template/spec/kubeadmConfigSpec/files/0/owner",
+						ValueMatcher: gomega.Equal("root:root"),
+					},
+					{
+						Operation:    "add",
+						Path:         "/spec/template/spec/kubeadmConfigSpec/files/0/permissions",
+						ValueMatcher: gomega.Equal("0600"),
+					},
+					{
+						Operation: "replace",
```

Comment:
The Patches look different now, with this replace, but the resulting KCP is the same.